src/main/java/org/usfirst/frc4904/robot/RobotMap.java

```diff
@@ -1,93 +1,168 @@
-    }
+    public static Motor intakeRollerMotor;
+    public static Motor funnelMotor;
+    public static Motor liftBeltMotor;
+    public static Motor runUpBeltMotor;
+    public static Motor flywheelMotorA;
+    public static Motor flywheelMotorB;
 
-    public static class Input {
-    }
+    public static CANCoder flywheelEncoder;
+    public static CANCoderConfiguration flywheelEncoderConfiguration;
+  }
 
-    public static class HumanInput {
-        public static class Driver {
-            public static CustomXbox xbox;
-        }
+  public static class Input {
+    public static CustomDigitalLimitSwitch limitSwitch;
+  }
 
-        public static class Operator {
-            public static CustomJoystick joystick;
-        }
+  public static class HumanInput {
+    public static class Driver {
+      public static CustomXbox xbox;
     }
 
-    public RobotMap() {
-        HumanInput.Driver.xbox = new CustomXbox(Port.HumanInput.xboxController);
-        HumanInput.Operator.joystick = new CustomJoystick(Port.HumanInput.joystick);
-
+    public static class Operator {
+      public static CustomJoystick joystick;
     }
+  }
+
+  public RobotMap() {
+    HumanInput.Driver.xbox = new CustomXbox(Port.HumanInput.xboxController);
+    HumanInput.Operator.joystick = new CustomJoystick(Port.HumanInput.joystick);
+
+    Component.intakeSolenoid = new SolenoidSubsystem(Port.Pneumatics.INTAKE_SOLENOID.buildDoubleSolenoid());
+    Component.flipperSolenoid = new SolenoidSubsystem(Port.Pneumatics.FLIPPER_SOLENOID.buildDoubleSolenoid());
+    Component.shooterAimSolenoid = new SolenoidSubsystem(Port.Pneumatics.SHOOTER_AIM_SOLENOID.buildDoubleSolenoid());
+
+    // TODO: FIX MOTOR TYPES
+    Component.intakeRollerMotor = new Motor("intakeRollerMotor", new CANTalonSRX(Port.CANMotor.INTAKE_ROLLER_MOTOR));
+    Component.funnelMotor = new Motor("funnelMotor", new CANTalonSRX(Port.CANMotor.INTAKE_FUNNEL_MOTOR));
+    Component.liftBeltMotor = new Motor("liftBeltMotor", new CANTalonSRX(Port.CANMotor.LIFT_BELT_MOTOR));
+    Component.runUpBeltMotor = new Motor("runUpBeltMotor", new CANTalonSRX(Port.CANMotor.RUN_UP_BELT_MOTOR));
+    Component.flywheelMotorA = new Motor("flywheelMotorA", new CANTalonFX(Port.CANMotor.FLYWHEEL_MOTOR_A));
+    Component.flywheelMotorB = new Motor("flywheelMotorB", new CANTalonFX(Port.CANMotor.FLYWHEEL_MOTOR_B));
+
+    Component.intake = new Intake(Component.intakeRollerMotor, Component.funnelMotor, Component.intakeSolenoid);
+    Component.indexer = new Indexer(Component.liftBeltMotor, Component.flipperSolenoid);
+
+    Component.flywheelEncoder = new CANCoder(Port.CAN.FLYWHEEL_ENCODER);
+    Component.flywheelEncoderConfiguration = new CANCoderConfiguration();
+    Component.flywheelEncoder.configAllSettings(Component.flywheelEncoderConfiguration);
+    Component.flywheel = new Flywheel(new CustomPIDController(PID.Flywheel.P, PID.Flywheel.I, PID.Flywheel.D, Component.flywheelEncoder); // TODO: BAAAAAAD CODE
```

Comment:
this should work once we have our own wrapper for `CANCoder` that implements `CustomEncoder`

Reply:
@NikhilSuresh24 wrote one, so you should be able to use that.